fvm/evm/emulator/emulator.go

```diff
@@ -179,7 +179,7 @@ func (bl *BlockView) RunTransaction(
 				if result != nil {
 					receipt = result.Receipt()
 				}
-				proc.evm.Config.Tracer.OnTxEnd(receipt, err)
+				proc.evm.Config.Tracer.OnTxEnd(receipt, result.ValidationError)
```

Comment:
maybe we use the nil check for results for error as well

Reply:
what do you mean?

Reply:
I also think there are 2 more places in the code that probably is facing the same problem

Reply:
yeah will look into it, a bit wip still

Reply:
Initially there was a problem that result is empty (e.g. backend errors)

Reply:
I guess I also need to check for `result != nil` although, I don't believe that was an issue in the first place, because I don't see how at this point result would be nil

Reply:
in case of BackendError on the commit (e.g. storage read over the limit) we return with empty results,